cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataEntity.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.metadata;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Entity representation for Metadata
+ */
+public class MetadataEntity {
+
+  public static final String NAMESPACE = "namespace";
+  public static final String APPLICATION = "application";
+  public static final String ARTIFACT = "artifact";
+  public static final String VERSION = "version";
+  public static final String DATASET = "dataset";
+  public static final String STREAM = "stream";
+  public static final String VIEW = "view";
+  public static final String TYPE = "type";
+  public static final String PROGRAM = "program";
+
+  private final List<KeyValue> details;
+
+  private MetadataEntity(List<KeyValue> details) {
+    this.details = new ArrayList<>(details);
+  }
+
+  /**
+   * Creates a {@link MetadataEntity} representing the given datasetName. Alternatively, you
+   * can also construct a {@link MetadataEntity} which is equivalent to DatasetId by providing the namespace using
```

Comment:
There is no `DatasetId` in API. it's confusing mentioning it here. Maybe simply just say `Creates a MetadataEntity representing the given dataset in the current namespace`